mitmproxy/utils/arg_check.py

```diff
@@ -0,0 +1,61 @@
+import sys
+
+DEPRECATED = """
+--cadir
+-Z
+--body-size-limit
+--stream
+--palette
+--palette-transparent
+-e
+--eventlog
+--follow
+--order
+--no-mouse
+--reverse
+--socks
+--http2-priority
+--np-http2-priority
+--no-websocket
+--websocket
+--spoof-source-address
+--upstream-bind-address
+--ciphers-client
+--ciphers-server
+--client-certs
+--no-upstream-cert
+--add-upstream-certs-to-client-chain
+--upstream-trusted-cadir
+--upstream-trusted-ca
+--ssl-version-client
+--ssl-version-server
+--no-onboarding
+--onboarding-host
+--onboarding-port
+--server-replay-use-header
+--norefresh
+--no-pop
+--replay-ignore-content
+--replay-ignore-payload-param
+--replay-ignore-param
+--replay-ignore-host
+--replace-from-file
+--nonanonymous
+--singleuser
+--htpasswd
+"""
+
+
+def check():
+    args = sys.argv[1:]
+
+    if "-U" in args:
+        print("-U is deprecated, please use --mode upstream:SPEC instead")
+
+    if "-T" in args:
+        print("-T is deprecated, please use --mode transparent instead")
+
+    if any([i in args for i in DEPRECATED.splitlines()]):
+        print("Looks like you used a flag which is deprecated.\n"
+              "To set an option please use `--set option_name=value` "
+              "To show all options and their default values use --options")
```

Comment:
Imagine someone calling this with four, five options. This would be more useful if we call out the actual flag that has been removed. Just make it a for-loop and print for each affected option? :)